File: Greg.java
import java.util.*;

public class Greg {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int x = sc.nextInt();   // n
        int y = sc.nextInt();   // m
        int z = sc.nextInt();   // k

        // Initial array
        long[] ques = new long[x];
        for (int i = 0; i < x; i++) {
            ques[i] = sc.nextLong();
        }

        // Operations
        int[][] op = new int[y][3];
        for (int i = 0; i < y; i++) {
            op[i][0] = sc.nextInt();
            op[i][1] = sc.nextInt();
            op[i][2] = sc.nextInt();
        }

        // Queries
        int[][] queries = new int[z][2];
        for (int i = 0; i < z; i++) {
            queries[i][0] = sc.nextInt();
            queries[i][1] = sc.nextInt();
        }

        // Difference array for operation counts
        long[] arr = new long[y + 1];

        for (int i = 0; i < z; i++) {
            arr[queries[i][0] - 1] += 1;
            arr[queries[i][1]] -= 1;
        }

        // Prefix sum to get how many times each operation is applied
        for (int i = 1; i < y; i++) {
            arr[i] += arr[i - 1];
        }

        // Difference array for the original array
        long[] res = new long[x + 1];

        for (int i = 0; i < y; i++) {
            long value = (long) op[i][2] * arr[i];

            res[op[i][0] - 1] += value;
            res[op[i][1]] -= value;
        }

        // Prefix sum to get final increments
        for (int i = 1; i <= x; i++) {
            res[i] += res[i - 1];
        }

        // Final answer
        for (int i = 0; i < x; i++) {
            System.out.print(ques[i] + res[i]);

            if (i != x - 1) {
                System.out.print(" ");
            }
        }

        System.out.println();

        sc.close();
    }
}

File: greg.py
import time
x,y,z=map(int,input().split())
ques=list(map(int,input().split()))
op=[list(map(int,input().split())) for _ in range(y)]
queries=[list(map(int,input().split())) for _ in range(z)]

start_time=time.time()
# print(arr)
# print(op)
# print(queries)

arr=[0 for _ in range(len(op)+1)]
print(arr)
for i in queries:
    arr[i[0]-1]+=1
    arr[i[1]]-=1
for j in range(1,len(op)):
    arr[j]+=arr[j-1]
print(arr)
res=[0 for _ in range(x+1)]
for k in range(len(op)):
    res[op[k][0]-1]+=op[k][2]*arr[k]
    res[op[k][1]]-=op[k][2]*arr[k]
for j in range(1,x+1):
    res[j]+=res[j-1] 
print(res)
result=list(map(lambda x,y:x+y,ques,res))
print(result)




print("Time taken: %s seconds" % (time.time() - start_time))